ribes_utils.py

```diff
@@ -0,0 +1,70 @@
+import itertools
```

Comment:
I think RIBES should be a `Scorer`. Could you move it there and have it follow the same format as, for example, `BleuScorer`? This would allow it to be re-used in other parts of the code.